### PR TITLE
Add confirmation to mpa approve

### PR DESCRIPTION
**Pull request by sfc-gh-elinardi**
This PR adds a confirmation prompt to `sanssh mpa approve` command. 
Currently, `sanssh mpa approve` would approve the request without informing the approver about what they're approving of.

Changes:
* Add a prompt to `sanssh mpa approve` command informing the approver about the details of the request, and asking them to enter `yes` or `no`. Approver can skip confirmation by specifying `--skip-confirmation`

Example:
Approving without `--skip-confirmation`
```
/sanssh ---targets=localhost --credential-source=flags --client-key ~/.sansshell/client2.key --client-cert ~/.sansshell/client2.pem mpa approve aacf28b4-55d31d53-1bc77c57            
MPA Request:
{
  "user": "user1",
  "method": "/LocalFile.LocalFile/Read",
  "message": {
    "@type": "type.googleapis.com/LocalFile.ReadActionRequest",
    "file": {
      "filename": "/etc/group3"
    }
  }
}
Would you like to approve the request? (yes/no): yes
Approving the request..
Approved /LocalFile.LocalFile/Read from user1
```
With `--skip-confirmation`
```
./sanssh --targets=localhost --credential-source=flags --client-key ~/.sansshell/client2.key --client-cert ~/.sansshell/client2.pem mpa approve--skip-confirmation f4c49abe-aea5fc54-733dc4a4
MPA Request:
{
  "user": "user1",
  "method": "/LocalFile.LocalFile/Read",
  "message": {
    "@type": "type.googleapis.com/LocalFile.ReadActionRequest",
    "file": {
      "filename": "/etc/group4"
    }
  }
}
Approved /LocalFile.LocalFile/Read from user1
```

Invalid input
```
./sanssh --targets=localhost --credential-source=flags --client-key ~/.sansshell/client2.key --client-cert ~/.sansshell/client2.pem mpa approve 7116c620-f16fa65c-36a3ac48            
MPA Request:
{
  "user": "user1",
  "method": "/LocalFile.LocalFile/Read",
  "message": {
    "@type": "type.googleapis.com/LocalFile.ReadActionRequest",
    "file": {
      "filename": "/etc/group5"
    }
  }
}
Would you like to approve the request? (yes/no): bla
Request is not approved. Exiting.
```

sanssh mpa approve --help
```
sanssh mpa approve --help              
approve <id> [--skip-confirmation]:
    Approves an MPA request with the specified ID.

        The --skip-confirmation flag can be used to bypass 
        the confirmation prompt, proceeding with the request approval.
  -skip-confirmation
        If true won't ask for confirmation
```

### CODE REVIEW

File: services/mpa/client/client.go [hunk 1]
```diff
@@ -113,20 +116,47 @@ func (*approveCmd) Usage() string {
...
+				break
+			}
+			if strings.ToLower(input) == "no" || strings.ToLower(input) == "n" {
+				fmt.Print("Request is not approved. Exiting.\n")
+				return subcommands.ExitSuccess
+			}
+			fmt.Println("Invalid input. Please enter 'yes' or 'no'")
+		}
+	}
+	fmt.Print("Approving the request..\n")
```
Comment (sfc-gh-srhodes) [inline services/mpa/client/client.go:158]: Maybe remove this line? I don't think it adds much value.

File: services/mpa/client/client.go [hunk 2]
```diff
@@ -113,20 +116,47 @@ func (*approveCmd) Usage() string {
...
+			input, err := reader.ReadString('\n')
+			if err != nil {
+				fmt.Println("Error reading input. Please try again.")
+				continue
+			}
+			input = strings.TrimSpace(input)
+			if strings.ToLower(input) == "yes" || strings.ToLower(input) == "y" {
+				break
+			}
+			if strings.ToLower(input) == "no" || strings.ToLower(input) == "n" {
```
Comment (sfc-gh-srhodes) [inline services/mpa/client/client.go:151]: I'd minorly prefer to exit on non-yes instead of requiring "no"

File: services/mpa/client/client.go [hunk 3]
```diff
@@ -103,7 +104,9 @@ func getAction(ctx context.Context, state *util.ExecuteState, c pb.MpaClientProx
 	return anyAction
 }
 
-type approveCmd struct{}
+type approveCmd struct {
+	skipConfirmation bool
+}
 
 func (*approveCmd) Name() string     { return "approve" }
 func (*approveCmd) Synopsis() string { return "Approves an MPA request" }
```
Comment (sfc-gh-srhodes) [inline services/mpa/client/client.go:112]: Could you mention the confirmation in the usage?

File: services/mpa/client/client.go [hunk 4]
```diff
@@ -113,20 +116,47 @@ func (*approveCmd) Usage() string {
...
 	c := pb.NewMpaClientProxy(state.Conn)
-	action := getAction(ctx, state, c, f.Args()[0])
+	action := getAction(ctx, state, c, id)
 	if action == nil {
 		return subcommands.ExitFailure
 	}
 
+	if !p.skipConfirmation {
+		// ask for confirmation
+		fmt.Printf("MPA Request:\n%s\n", protojson.MarshalOptions{UseProtoNames: true, Multiline: true}.Format(action))
```
Comment (sfc-gh-srhodes) [inline services/mpa/client/client.go:138]: I forgot that we were already fetching this. What do you think of unconditionally printing the request?

**Review by sfc-gh-srhodes — CHANGES_REQUESTED** [review]

File: services/mpa/client/client.go (hunk 4 above)
Comment (sfc-gh-elinardi) [reply services/mpa/client/client.go:138]: I really like that idea.

File: services/mpa/client/client.go (hunk 1 above)
Comment (sfc-gh-elinardi) [reply services/mpa/client/client.go:158]: Removed

File: services/mpa/client/client.go (hunk 2 above)
Comment (sfc-gh-elinardi) [reply services/mpa/client/client.go:151]: Makes sense. Updated to exit non non-yes

File: services/mpa/client/client.go (hunk 3 above)
Comment (sfc-gh-elinardi) [reply services/mpa/client/client.go:112]: Added

File: services/mpa/client/client.go (hunk 4 above)
Comment (sfc-gh-elinardi) [reply services/mpa/client/client.go:138]: Changed to always print the request

**Review by sfc-gh-srhodes — APPROVED** [review]
Thanks!

**Pull request merged**